Algorithmia/__main__.py

```diff
@@ -0,0 +1,182 @@
+import sys
+import os
+import json
+import Algorithmia
+from Algorithmia.CLI import CLI
+import argparse
+
+
+#CLI app to allow a user to run algorithms and manage data collections
+
+usage = """CLI for interaction with Algorithmia\n
+Usage:\n
+algo [<cmd>] [options] [<args>...]\n
+algo[<cmd>] [--help | --version]\n\n
+
+General commands include:\n
+  auth	configure authentication\n\n
+
+  Algorithm commands include:\n
+    run	Runs an algorithm\n
+    clone	Clones an algorithm source\n\n
+
+  Data commands include:\n
+    ls list the contents of a data directory\n
+    mkdir create a data directory\n
+    rmdir remove a data directory\n
+    rm remove a file from a data directory\n
+    cp copy file(s) to or from a data directory\n
+    cat concatinate and print file(s) in a data directory\n\n
+
+  Global options:\n
+    --help\n
+    --profile <name>\n\n
+"""
+
+def main():
+	parser = argparse.ArgumentParser('algo', description = usage)
+
+	subparsers = parser.add_subparsers(help = 'sub cmd',dest = 'cmd')
+
+	parser_auth = subparsers.add_parser('auth', help = 'save api key and api address for profile')
+	parser_auth.add_argument('--profile', action = 'store', type = str, default = 'default')
+
+	parser_clone = subparsers.add_parser('clone', help = 'clone <algo>')
+	parser_clone.add_argument('algo')
+	parser_clone.add_argument('--profile', action = 'store', type = str, default = 'default')
+
+	#parse options for the run command
+	parser_run = subparsers.add_parser('run', help = 'algo run <algo> [input options] <args..> [output options]')
+
+	parser_run.add_argument('algo')
+	parser_run.add_argument('-d','--data', action = 'store_true', help = 'detect input type')
+	parser_run.add_argument('-t','--text', action = 'store_true', help = 'treat input as text')
+	parser_run.add_argument('-j','--json', action = 'store_true', help = 'treat input as json data')
+	parser_run.add_argument('-b','--binary', action = 'store_true', help = 'treat input as binary data')
+	parser_run.add_argument('-D','--data-file', action = 'store_true', help = 'spesify a path to an input file')
+	parser_run.add_argument('-T','--text-file', action = 'store_true', help = 'spesify a path to a text file')
```

Comment:
behavior of -T is not consistent with generated docs. Should behave like -o which can be anywhere in the command.

Reply:
input options can now appear anywhere in the command but one must be specified so "algo run -d "test" util/echo" is valid but " algo run util/echo "test" " is not